llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -586,6 +586,13 @@ static string_vector saveDeviceImageProperty(
             NameInfoPair.first, llvm::util::PropertyValue(Data, DataBitSize)));
       }
     }
+
+    if (ImgPSInfo.IsEsimdKernel) {
+      std::map<StringRef, bool> RMEntry = {{"isEsimdImage", true}};
+      PropSet.add(llvm::util::PropertySetRegistry::SYCL_IS_ESIMD_IMAGE,
+                  RMEntry);
+    }
```

Comment:
This is a whole property set for a single `string, boolean` pair and as I can see, this is not the first one such set:

https://github.com/intel/llvm/blob/557b564049afbb8bb4d7a6bf22ff15fb0559b3e3/llvm/tools/sycl-post-link/sycl-post-link.cpp#L544-L549

What do you think about creating a single generic property set for boolean properties? This will slightly reduce amount of code and it will also reduce amount of temporary files created by the compiler

Tagging @kbobrovs here as well

Reply:
@AlexeySachkov, I don't feel qualified enough to judge your proposal. I don't know this part of the code very well.

Reply:
> What do you think about creating a single generic property set for boolean properties?

this seems like a good idea to me with some tweaks. But I think merging this with DeviceLibReqMask would be confusing, as the device lib mask is not a boolean, but a set of booleans. So basically, `SYCL/is ESIMD image` could just be renamed to something like [SYCL/Binary Image Properties] which could contain all binary image-related properties, not only booleans. Having a bit mask instead of separate boolean properties wouldn't have much perf impact, but would reduce readability, IMO.

> it will also reduce amount of temporary files created by the compiler

No sure about his one. @AlexeySachkov - could you elaborate please?

Reply:
> But I think merging this with DeviceLibReqMask would be confusing, as the device lib mask is not a boolean, but a set of booleans. 

Technically that property is absolutely the same as `SYCL/is ESIMD image` since each property set is a set by its nature, but for `llvm::util::PropertySetRegistry::SYCL_DEVICELIB_REQ_MASK` we only store a single pair within it - that's why I was referring to it as to a simple boolean property.

Note: I'm not asking to do such refactoring right now in scope of this PR, just highlighted a potential room for improvement

> > it will also reduce amount of temporary files created by the compiler
>
> No sure about his one. @AlexeySachkov - could you elaborate please?

Please disregard this. By some reason I though that we emit separate `.prop` file for each propety set, but this is not true - we put all property set related to a particular device image and store then into a single `.prop` file

Reply:
I still think it would be good not to create a separate category per single property. Once out, all categories/property names become legacy which we need to maintain backward compatibility with.

Reply:
Maybe 'SYCL/misc' would also work.